core/src/main/java/org/elasticsearch/index/query/QueryBuilder.java

```diff
@@ -30,6 +36,57 @@
 
 public interface QueryBuilder<QB extends QueryBuilder> extends NamedWriteable<QB>, ToXContent {
 
+    public static enum Operator implements Writeable {
+        OR(0), AND(1);
+        
+        private final int ordinal;
+
+        static final Operator PROTOTYPE = OR;
+
+        private Operator(int ordinal) {
+            this.ordinal = ordinal;
+        }
+        
+        public BooleanClause.Occur toBooleanClauseOccur() {
+            switch (this) {
+                case OR:
+                    return BooleanClause.Occur.SHOULD;
+                case AND:
+                    return BooleanClause.Occur.MUST;
+                default:
+                    throw Operator.newOperatorException(this.toString());
+            }
+        }
```

Comment:
maybe we can add a static method like this and make the odd PROTYPE private, or maybe we even want to remove the constant for it at that point?

 ```
public static Operator readFrom(StreamInput in) throws IOException {
    return PROTOTYPE.readFrom(in);
}
```

To be clear I wasn't against your previous static method, that was fine, it's just that I would want the enum to explicitly implement either Writeable or Streamable, which it does now ;)